### ISSUE TITLE
change_url action should not allow to change the path for full-domain apps

### ISSUE THREAD

**Issue by tituspijean** [YunoHost/issues#1798]
YunoHost v4.24

Sorry for the title, feel free to make it more clear.

There is a use case with changing app URL that does not seem to be correctly handled. Some apps can only be installed at the root of the domain, although the system still allows to change its URL and add a path after installation.

Example with Cryptpad:

![image](https://user-images.githubusercontent.com/8769166/118010407-4963d600-b34f-11eb-96be-a5ac2dd0672f.png)

Cryptpad's script, as of now, simply ignores the path part and forces `/`. ***However*, the path input by the user is still changed in the app settings.** Here, after changing the URL like displayed in the screenshot, the NGINX config path is still `/` but my Cryptpad settings are: 

```
...
domain: pad.pijean.ovh
final_path: /var/www/cryptpad
id: cryptpad
install_time: 1620809457
nodejs_version: '14'
path: /t
path_url: /
port: '3002'
porti: '3003'
```

(notice the `path` and `path_url`, but that may be a different issue)
(I noticed the issue after trying to put `/t` as path for the lolz, and while reverting back I got the screenshot error.)

To summarize, rwo issues are combined in here:
- Domain-only apps can still have their path changed after installation
- YunoHost settings are altered at the system-level, not by the change_url script, leading to a possible mismatch between NGINX and stored settings when change_url script ignores the new path.

**Comment by alexAubin** [comment]
Should be fixed by https://github.com/YunoHost/yunohost/commit/80f4b892e65b6838146bc520da8feb7991327a32

**Comment by tituspijean** [comment]
Bug is still present on v4.3.6.2. 😢 

Example:
1. install nitter on nitter.local
2. change its url to nitter.local/nitter
3. even though its script ignores the path change, the new url is displayed in the webadmin
(the app is broken too, but that's another issue I guess)
